dev_docs/07_visual_testing_guide.md

```diff
@@ -0,0 +1,128 @@
+
+      - `renderComponent(component, props, slots)`: Renders the specified component with given props and slots in the visual testing playground.
+      - `takeSnapshot(name, options)`: Takes a Percy snapshot with the given name and options.
+      - `click(selector)`, `hover(selector)`, `scrollToPos(selector, scrollOptions)`, `waitFor(selector)`, `delay(time)`: Utility functions for simulating user interactions.
+
+2. **Write Tests**:
+   - Use the utility functions to render components and take snapshots. For example:
+
+     ```javascript
+     describe.visual('KButton Visual Tests', () => {
+       it('Sample test for KButton', async () => {
+         await renderComponent('KButton', { text: 'Raised Button', appearance: 'raised-button' });
+         await takeSnapshot('KButton - Raised Button', { widths: [375, 520] });
+       });
+     });
+     ```
+     Note that the `widths` parameter passed to the `takeSnaphot` function is a part of Percy CLI's snapshot options. For a full list of available options, refer the [Percy documentation](https://www.browserstack.com/docs/percy/take-percy-snapshots/snapshots-via-scripts#per-snapshot-configuration).
+
+   - For rendering complex commponents, refer to the following:
+
+      - **Example with slots:** For components that involve slots, you can render them with `renderComponent` by passing the slot structure using element and elementProps. You can pass multiple slots at once.
+
+           ```javascript
+            await renderComponent('KIconButton', { icon: 'add' }, {
+              menu: { // slot named #menu
+                element: 'KDropdownMenu',
+                elementProps: {
+                  items: ['Option 1', 'Option 2'],
+                },
+              },
+            }); 
+          ```
+
+          **Note:** Use `'default'` key for passing default slots, with the HTML content specified using `innerHTML` prop. Checkout [`KButton.spec.js`](../lib/buttons-and-links/__tests__/KButton.spec.js) for reference.
+
+      - **Example involving more complex component structures:** When dealing with more complex component structures, it's recommended to create a dedicated Vue component for visual testing purposes. Add all the use cases in a Vue file and then render the custom component using the `renderComponent` function. 
+
+          ```javascript
+            await renderComponent('CustomVueComponent');
+          ```
+
+          This approach ensures that all necessary child components and slots are correctly set up and rendered.
+
+   - Make sure to use `describe.visual` or `it.visual` instead of the default notations for writing test blocks containing visual tests so as to prevent any unexpected behavior. These custom blocks add a [Visual] tag to the test name whose presence or absence are then checked using a regex pattern based on the type of tests executed. This implementation helps us to determine which test blocks should be executed by Jest and which ones should be skipped.
```

Comment:
I would suggest to be more specific "of the default `describe` and `it` funtions".

We should add that anything inside describe.visual will not be executed as unit test

Reply:
Sure, will make the required changes 👍